src/sagas/RootSaga.js

```diff
@@ -24,93 +24,75 @@ import {
   deleteProfilePhotoSuccess,
   deleteProfilePhotoFailure,
   deleteProfilePhotoReset,
+  resetDrafts,
 } from '../actions/ProfileActions';
 
-import * as ProfileApiService from '../services/ProfileApiService';
-
-
-const PROP_TO_STATE_MAP = {
-  fullName: 'name',
-  userLocation: 'country',
-  education: 'levelOfEducation',
-  socialLinks: socialLinks => socialLinks.filter(({ socialLink }) => socialLink !== null),
-};
+// Selectors
+import { handleSaveProfileSelector } from '../selectors/ProfilePageSelector';
 
-export const mapDataForRequest = (props) => {
-  const state = {};
-  Object.keys(props).forEach((prop) => {
-    const propModifier = PROP_TO_STATE_MAP[prop] || prop;
-    if (typeof propModifier === 'function') {
-      state[prop] = propModifier(props[prop]);
-    } else {
-      state[propModifier] = props[prop];
-    }
-  });
-  return state;
-};
+// Services
+import * as ProfileApiService from '../services/ProfileApiService';
 
 export function* handleFetchProfile(action) {
   const { username } = action.payload;
+  const currentUsername = yield select(state => state.authentication.username); // eslint-disable-line
 
   try {
     yield put(fetchProfileBegin());
 
-    const profile = yield call(
-      ProfileApiService.getProfile,
-      username,
-    );
-    const preferences = yield call(
-      ProfileApiService.getPreferences,
-      username,
-    );
-    profile.certificates = yield call(
-      ProfileApiService.getCourseCertificates,
-      username,
-    );
-
-    yield put(fetchProfileSuccess(profile));
-    yield put(receivePreferences(preferences));
+    const calls = [
+      call(ProfileApiService.getAccount, username),
+      call(ProfileApiService.getCourseCertificates, username),
+    ];
+
+    if (username === currentUsername) {
+      calls.push(call(ProfileApiService.getPreferences, username));
+    }
+
+    const result = yield all(calls);
+
+    if (result.length > 2) {
+      const [account, certificates, preferences] = result;
+      yield put(fetchProfileSuccess(account, preferences, certificates));
+    } else {
+      const [account, certificates] = result;
+      yield put(fetchProfileSuccess(account, { visibility: {} }, certificates));
```

Comment:
This may be a little weird, recognize the need tho

Reply:
Yeah, I couldn't figure out a more elegant way to handle it.